lib/active_shipping/location.rb

```diff
@@ -73,10 +73,11 @@ def self.from(object, options = {})
 
       ATTRIBUTE_ALIASES.each do |attribute, aliases|
         aliases.detect do |sym|
-          value = if hash_access
-            object[sym]
-          elsif object.respond_to?(sym)
-            object.send(sym)
+          value = object[sym] if hash_access
+          if !value &&
+            object.respond_to?(sym) &&
+            (!hash_access || !Hash.public_instance_methods.include?(sym))
+            value = object.send(sym)
           end
```

Comment:
I'm wondering if we organize the code a little simpler by extracting the `hash_access` and `!hash_access`
behaviour in their own class methods, perhaps something like this?

Also *second point*:

Seems like `respond_to?` only checks for public_methods, so we could safely use `public_send` here rather then `send`.

```ruby
    class << self
      def from(object, options = {})
        return object if object.is_a?(ActiveShipping::Location)

        attributes = {}

        accessor = if object.respond_to?(:[])
          method(:hash_accessor)
        else
          method(:public_method_accessor)
        end

        ATTRIBUTE_ALIASES.each do |attribute, aliases|
          aliases.detect do |sym|
            value = accessor.(object, sym)
            attributes[attribute] = value if value
          end
        end

        attributes.delete(:address_type) unless ADDRESS_TYPES.include?(attributes[:address_type].to_s)

        new(attributes.update(options))
      end

      def hash_accessor(object, sym)
        value = object[sym]

        if (
          !value &&
          object.respond_to?(sym) &&
          !Hash.public_instance_methods.include?(sym)
        )
          value = public_method_accessor(sym)
        end

        value
      end

      def public_method_accessor(object, sym)
        object.public_send(sym)
      end
```